Cookbook/Proposals/AdditionalRulesForDanger.md

```diff
@@ -0,0 +1,133 @@
+With the exception of the empty space component (even though it has its own NVL equivalent), every other pre-NVL component used inside Renderer should trigger a warning.
+
+The following regex currently yields 1333 results in 216 files: `Component\.(?!EmptySpace)\b`.
+The sheer number of warnings immediately excludes SwiftLint; Danger would be our only option.
+
+Nevertheless, this would require that everyone relied exclusively on NVL components and I am not sure this is the case right now.
+
+
+### Mutating Current in `setUp()` but not restoring it in `tearDown()`
+
+The rule is simple: every object (susceptible to global mutation) that is mutated in `setUp()` must be restored in `tearDown()`.
+Mutations in `Current` in particular are the prime suspects of a number of flaky tests that have been recently affecting our test suite.
+
+In order to detect this problem we have to use Danger since SwiftLint only accepts regular expressions.
+We would have to whitelist every test file (`.*Tests.*`) and use Danger to manually parse the test line-by-line until the `setUp()` function was found. 
+
+If found, we need to make a counter that increases by one each time a curly bracket and does the reverse when a closing curly bracket appears. Once the counter reaches zero again, then we've successfully delimited the contents of the `setUp()` function and we'd use the following regex to detect changes in Current: `Current\..*\ = `.
+
+The algorithm is then be executed once more but for the `tearDown()` method instead.
+If the number of changes inside each function matches, then no issue would be found. If not, then Current appeared to have been mutated but not restored; we'd issue a warning for the lines in which the algorithm detected a change and alert the developers in GitHub.
+
+Nevertheless, this rule is rather naïve because we have no actual way of detecting if Current was reset or not; that would require compiling and runniing the actual Swift code. Therefore, this approach is not a silver bullet or anything remotely similar but it would still be useful for one use case in particular: should the developer forget to restore Current, this rule will alert us of this situation.
+
+### Translation errors
+
+Translations are particularly hard to catch during PR code reviews, especially if a lot of new phrases are added at once.
+While, in theory, we have a fastlane set up to deal with these translations there is always the risk that developers mistype the localization key when setting up a new static var.
+
+As such, I'd like to suggest another rule for Danger that would:
+    1. detect new static String variables in every Localizable.swift file.
+    2. if we detected a new static variable we would then scan the localization string file (`Babylon/Supporting Files/Localization/en.lproj/Localizable.strings`) and look for that key
+
+We would need to whitelist `Localizable.*\.swift` in order to make this work.
+
+### Feedbacks must have the `when` prefix
```

Comment:
I'm not sure about this one, we have Feedbacks which start their effects when the system is initialised and not depending on a certain state so I think it would be misleading enforcing this prefix to them.

Reply:
@dmcrodrigues then it can be called `whenInitialized` or `whenStarted` or something like that? I'm +1 for enforcing common naming patterns in general. Similarly it would be nice to enforce common pattern in Actions/Events, i.e. that they start or should not start with `did` and they don't match state names. But it might be trickier as we are dealing with natural language processing here and a simple regex can be too limiting.